metrics/recorder_test.go

```diff
@@ -180,11 +181,22 @@ func TestRecorder_ConcurrentSafety(t *testing.T) {
 	go thread()
 	go thread()
 
-	<-ch
-	<-ch
+	for finishedThreads := 0; finishedThreads < 2; {
+		select {
+		case <-time.After(time.Second):
```

Comment:
you're waiting for 1 second twice, is it intended? otherwise you might want to save the channel in a var prior to the loop